### PR TITLE
feat: Add analyzer for missing packages when using ProgressRing or MPE

### LINKED ISSUE #17026: Fail the build on uses of MediaPlayer and ProgressRing if related packages cannot be found
### What would you like to be added

The XAML Generator should detect uses of those types and fail if the related support packages are not included in the build.

### Why is this needed

Better initial experience and avoid runtime-only failures

### For which platform

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #17026

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.Analyzers/UnoMissingAssemblyAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+#pragma warning disable RS2008 // Enable analyzer release tracking
+		"Uno0007",
+#pragma warning restore RS2008 // Enable analyzer release tracking
+		Title,
+		MessageFormat,
+		Category,
+		DiagnosticSeverity.Warning,
+		isEnabledByDefault: true,
+		helpLinkUri: "https://aka.platform.uno/UNO0007"
```
Comment (Youssef1313) [inline src/Uno.Analyzers/UnoMissingAssemblyAnalyzer.cs:28]: @jeromelaban The url needs to be added to the aka database

**Comment by Youssef1313** [comment]
```
/Users/runner/work/1/s/src/SolutionTemplate/5.1/uno51blank/uno51blank.Skia.Gtk/Uno.UI.SourceGenerators/Uno.UI.SourceGenerators.BindableTypeProviders.BindableTypeProvidersSourceGenerator/BindableMetadata.g.cs(8275,45): error Uno0007: Using 'MediaPlayer' requires 'Uno.WinUI.MediaPlayer.Skia.Gtk' NuGet package to be referenced (https://aka.platform.uno/UNO0007) [/Users/runner/work/1/s/src/SolutionTemplate/5.1/uno51blank/uno51blank.Skia.Gtk/uno51blank.Skia.Gtk.csproj]

```

I'm not sure what should we do for `BindableTypeProvidersSourceGenerator`

File: src/Uno.Analyzers/UnoMissingAssemblyAnalyzer.cs (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.Analyzers/UnoMissingAssemblyAnalyzer.cs:28]: Done!

**Review by jeromelaban — APPROVED** [review]

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.3

**Review by carldebilly — APPROVED** [review]
Awesome!

**Pull request merged**